### PR TITLE
Allow a nodeSelector to be specified on install

### LINKED ISSUE #54: Support PodNodeSelector
I'm happy to take a stab at contributing this, but wanted to make sure it was something you'd accept first.

I'd like to be able to instal all codefresh resources on a specific set of nodes. This is done (one way) via a node selector. https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector

The idea would be to run `venona install --node-selector "foo:bar"` which would then install all resources on only nodes with the label foo and value bar.

### PR DESCRIPTION
**Pull request by kpurdon**
Allows a nodeSelector to be specified on install by passing `--kube-node-selector="foo:bar"` to `venona install`.

resolves #54

### CODE REVIEW

**Comment by kpurdon** [comment]
I've taken this first pass at getting the basics as far as I can tell in place. I believe this covers the 3 services installed by venona, but it was unclear to me where the ephemeral services (created per build) are created. Those, ideally are also subject to this addition.

cc @olegsu ... since there are no contribution guidelines I'm just pinging you since you seem to be recently active on the project. Please let me know if I should reach out to others for review.

I additionally don't see any instructions on how to test or validate these changes. Any guidance there would be appreciated.

**Comment by olegsu** [comment]
Hey @kpurdon thanks for the PR.
Your change actually will affect the `node-selector` of all venona services, which is great.
To be able to affect as well the ephemeral services (2 pods per build in our case), we need to have another flag for that, which should be part of the registration request to Codefresh backend https://github.com/codefresh-io/venona/blob/master/venonactl/pkg/plugins/runtime-environment.go#L44. Atm this flag is not supported in our backend system, we will add it to our backlog.

venona is build of 2 parts( server + cli ), and you probably want to test the installation part, so to compile the cli, just `cd venonactl && make build`  it will add to your `venonactl` to `/usr/local/bin/venonactl` as mention in https://github.com/codefresh-io/venona/blob/master/venonactl/hack/build.sh#L3, so you can run it by `venonactl install --kube-namespace ...`

Regardless, I will make sure to have clear instructions and contribution guidelines.
Thanks for the feedback

**Comment by kpurdon** [comment]
@olegsu thanks! I'll hold on this PR until the CodefreshAPI work is complete, then I'll add that here. It's really the key piece for my needs anyway.

If there is anything I can do to help with that let me know.

**Comment by olegsu** [comment]
Hey @kpurdon 
We support now `nodeSelector` as part of the runtime environment registration API request.
To do so we need to use the latest `codefresh-io/go-sdk`(0.15.0) module (`go get -u ...`)
Pass it to https://github.com/codefresh-io/venona/blob/f703dd58881ab43b094ea15ce4c620d506409f33/venonactl/pkg/codefresh/cfapi.go#L163 
I think that a flag called `build-node-selector` should do the job.

**Comment by kpurdon** [comment]
@olegsu great news, thanks for the fast turnaround on that. I'll get back to this ASAP (hopefully this coming week).

**Comment by kpurdon** [comment]
@olegsu looks like v0.15.0 hasn't been released yet for the go-sdk. Can you publish a release there so I can update the module here? Thanks!

File: venonactl/go.mod [hunk 1]
```diff
@@ -40,3 +37,5 @@ require (
 	k8s.io/klog v0.0.0-20181108234604-8139d8cb77af // indirect
 	sigs.k8s.io/yaml v1.1.0 // indirect
 )
+
+replace git.apache.org/thrift.git => github.com/apache/thrift v0.0.0-20181218151757-9b75e4fe745a
```
[the remark this answers is not in the document]
Comment (olegsu) [reply venonactl/go.mod]: will check that

**Comment by olegsu** [comment]
> @olegsu I was able to build this and test an install on my infra. It doesn't seem that the per-build resources are getting the nodeSelector. Am I missing anything obvious?

What is your install cmd?
Will run it locally

**Comment by kpurdon** [comment]
Given a cluster w/ a node pool with the label `role: ancillary` and a namespace `codefresh-runtime` I run `make build-linux` and then `./venonactl-linux install --kube-node-selector="role: ancillary" --build-node-selector="role: ancillary" --kube-namespace=codefresh-runtime`

File: venonactl/go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (kpurdon) [reply venonactl/go.mod]: I also had quite a few issues building w/ go 1.13 due to some changes in modules validation. I didn't want to dig into that so am just building w/ 1.12.

**Comment by olegsu** [comment]
@kpurdon Sorry for the delay,
Let's try to pass it as `venona install --kube-build-selecotr LABEL_A=V_1 --kube-build-selecotr LABEL_B=V_2` than passing it to Codefresh API as `map[string]string` using the SDK, (was updated to v0.16.0)

**Comment by kpurdon** [comment]
@olegsu no worries ... just to be clear the `map[string]string` would be `[]{"LABEL_A":"V_1","LABEL_B":"V_2"}`?

**Comment by olegsu** [comment]
@kpurdon Yes

**Comment by kpurdon** [comment]
@olegsu I made those changes and it still doesn't seem that the build resources are getting the node selector. Anything I can to help debug?

**Comment by olegsu** [comment]
Thank @kpurdon ,
Can you print the registration requests as pure json? (masking credentials)

**Comment by kpurdon** [comment]
@olegsu I found the fix 🎉 ... I was able to get this fully working. All resources had the nodeSelector. Let's call this RFR!

**Comment by olegsu** [comment]
@kpurdon that's great!
I will be able to test it during Sunday, please also update `package.json` version, bump minor, please.

**Comment by olegsu** [comment]
@kpurdon please make sure to update the templates files
`go generate ${PWD}/hack/generate.go`

**Comment by kpurdon** [comment]
> please make sure to update the templates files go generate ${PWD}/hack/generate.go

@olegsu done

**Comment by olegsu** [comment]
lgtm

**Pull request merged**